cmake/Deps_Unix.cmake

```diff
@@ -79,30 +79,25 @@ endif ()
 if (WITH_ZIM)
     if (APPLE)
         # ICU from homebrew is "key-only", we need to manually prioritize it -> see `brew info icu4c`
-        set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/local/opt/icu4c@76/lib/pkgconfig:/opt/homebrew/opt/icu4c@76/lib/pkgconfig:/usr/local/opt/icu4c/lib/pkgconfig:/opt/homebrew/opt/icu4c/lib/pkgconfig")
+        # And we needs to find the correct one if multiple versions co exists.
+        set(ENV{PATH} "$ENV{PATH}:/usr/local/bin/:/opt/homebrew/bin") # add brew command into PATH
+        execute_process(
+                COMMAND sh -c [=[brew --prefix $(brew deps libzim | grep icu4c)]=]
+                OUTPUT_VARIABLE ICU_REQUIRED_BY_ZIM_PREFIX
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+                COMMAND_ERROR_IS_FATAL ANY)
+        message(STATUS "Found correct homebrew icu path -> ${ICU_REQUIRED_BY_ZIM_PREFIX}")
+        set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:${ICU_REQUIRED_BY_ZIM_PREFIX}/lib/pkgconfig")
```

Comment:
The PKG_CONFIG_PATH or CMAKE_PREFIX_PATH to be updated so that libzim can find icu4c via pkgconfig (see `brew info icu4c`)